snippets/base/templates/base/includes/snippet_js.html

```diff
@@ -355,19 +358,45 @@
         }
     }
 
-    // Download the user's country using the geolocation service.
-    function downloadUserCountry() {
-        $script(GEO_URL, 'geo');
-        $script.ready('geo', function() {
-            try {
-                gSnippetsMap.set('geoCountry', geoip_country_code());
-                gSnippetsMap.set('geoLastUpdated', new Date());
-            } catch (e) {
-                // Most likely failed to load JS file. Continue on without us,
-                // we'll try again next time.
-            }
-        });
-    }
+    {% if client.channel == 'release' %}
+      function downloadUserCountry() {
+          var GEO_URL = "{{ settings.OLD_GEO_URL }}";
+          $script(GEO_URL, 'geo');
+          $script.ready('geo', function() {
+              try {
+                  gSnippetsMap.set('geoCountry', geoip_country_code());
+                  gSnippetsMap.set('geoLastUpdated', new Date());
+              } catch (e) {
+                  // Most likely failed to load JS file. Continue on without us,
+                  // we'll try again next time.
+              }
+          });
+      }
+    {% else %}
+      // Download the user's country using the geolocation service.
+      // Please do not directly use this code or Snippets key.
+      // Contact MLS team for your own credentials.
+      // https://location.services.mozilla.com/contact
+      function downloadUserCountry() {
+          var GEO_URL = "{{ settings.GEO_URL }}";
+          var request = new XMLHttpRequest();
+          request.onreadystatechange = function() {
```

Comment:
maybe add ``request.timeout = 60000; // timeout after 60 seconds``

Networks are unreliable and I'd always add timeouts to any network request, so they don't stay around forever.